### PR TITLE
[GO] - Add support for v1.23

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Template**
* Go, Go-Postgres

**Description**
* Updated respective `devcontainer-template.json` files and `readme.md` files for the templates.

**Checklist**
* [x] Checked that applied changes work as expected

### CODE REVIEW

File: src/go/README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Develop Go based applications. Includes appropriate runtime args, Go, common too
 
 | Options Id | Description | Type | Default Value |
 |-----|-----|-----|-----|
-| imageVariant | Go version: | string | 1.22-bookworm |
```
Comment (amethystens) [inline src/go/README.md:10]: 1.23-bookworm

File: src/go-postgres/devcontainer-template.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "id": "go-postgres",
-    "version": "4.0.2",
```
Comment (amethystens) [inline src/go-postgres/devcontainer-template.json:3]: 4.0.3

File: src/go/devcontainer-template.json [hunk 3]
```diff
@@ -12,13 +12,15 @@
...
                 "1-bookworm",
+				"1.23-bookworm",
                 "1.22-bookworm",
                 "1.21-bookworm",
                 "1-bullseye",
+				"1.23-bullseye",
                 "1.22-bullseye",
                 "1.21-bullseye"
             ],
-            "default": "1.22-bookworm"
```
Comment (amethystens) [inline src/go/devcontainer-template.json:21]: 1.23

File: src/go-postgres/README.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ Use and develop Go + Postgres applications. Includes appropriate runtime args, G
 
 | Options Id | Description | Type | Default Value |
 |-----|-----|-----|-----|
-| imageVariant | Go version: | string | 1.22-bookworm |
```
Comment (amethystens) [inline src/go-postgres/README.md:10]: 1.23

File: src/go-postgres/devcontainer-template.json [hunk 5]
```diff
@@ -12,13 +12,15 @@
...
                 "1-bookworm",
+				"1.23-bookworm",
                 "1.22-bookworm",
                 "1.21-bookworm",
                 "1-bullseye",
+				"1.23-bullseye",
                 "1.22-bullseye",
                 "1.21-bullseye"
             ],
-            "default": "1.22-bookworm"
```
Comment (amethystens) [inline src/go-postgres/devcontainer-template.json:21]: 1.23

File: src/go/README.md (hunk 1 above)
Comment (amethystens) [inline src/go/README.md:10]: 1.23

File: src/go/devcontainer-template.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
     "id": "go",
-    "version": "4.0.2",
```
Comment (amethystens) [inline src/go/devcontainer-template.json:3]: 4.0.3

File: src/go/devcontainer-template.json (hunk 3 above)
Comment (amethystens) [inline src/go/devcontainer-template.json:21]: 1.23

**Review by amethystens — APPROVED** [review]

File: src/go/devcontainer-template.json (hunk 6 above)
Comment (amethystens) [inline src/go/devcontainer-template.json:3]: 4.0.3

File: src/go/devcontainer-template.json (hunk 3 above)
Comment (amethystens) [inline src/go/devcontainer-template.json:21]: 1.23

File: src/go-postgres/README.md (hunk 4 above)
Comment (amethystens) [inline src/go-postgres/README.md:10]: 1.23

**Review by amethystens — CHANGES_REQUESTED** [review]

File: src/go-postgres/README.md [hunk 7]
```diff
@@ -7,7 +7,7 @@ Use and develop Go + Postgres applications. Includes appropriate runtime args, G
...
+| imageVariant | Go version: | string | 1.23-bookworm |
```
Comment (samruddhikhandale) [inline src/go-postgres/README.md:10]: Once this PR is merged, https://github.com/devcontainers/action will run automatically ; which opens a PR that updates the README file based on `devcontainer-feature.json` and `NOTES.md`.

Hence, can we revert the changes to this file to avoid merge conflicts? Thanks!

File: src/go-postgres/devcontainer-template.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
     "id": "go-postgres",
-    "version": "4.0.2",
+    "version": "4.0.3",
```
Comment (samruddhikhandale) [inline src/go-postgres/devcontainer-template.json:3]:
```suggestion
    "version": "4.1.0",
```

File: src/go/README.md [hunk 9]
```diff
@@ -7,7 +7,7 @@ Develop Go based applications. Includes appropriate runtime args, Go, common too
...
+| imageVariant | Go version: | string | 1.23-bookworm |
```
Comment (samruddhikhandale) [inline src/go/README.md:10]: Same comment here

File: src/go/devcontainer-template.json [hunk 10]
```diff
@@ -1,6 +1,6 @@
...
+    "version": "4.0.3",
```
Comment (samruddhikhandale) [inline src/go/devcontainer-template.json:3]: same here

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: src/go/devcontainer-template.json (hunk 10 above)
Comment (gauravsaini04) [reply src/go/devcontainer-template.json]: changed

File: src/go/README.md (hunk 9 above)
Comment (gauravsaini04) [reply src/go/README.md:10]: changed

File: src/go-postgres/README.md (hunk 7 above)
Comment (samruddhikhandale) [reply src/go-postgres/README.md:10]: @gauravsaini04 Can you help resolve this?

File: src/go-postgres/devcontainer-template.json [hunk 11]
```diff
@@ -12,13 +12,15 @@
             "description": "Go version:",
             "proposals": [
                 "1-bookworm",
+				"1.23-bookworm",
```
Comment (samruddhikhandale) [inline src/go-postgres/devcontainer-template.json:15]: Let's remove 1.21 as well?

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

**Review by samruddhikhandale — APPROVED** [review]
